poetry/console/commands/add.py

```diff
@@ -184,3 +183,21 @@ def handle(self):
             self.poetry.file.write(original_content)
 
         return status
+
+    def remove_existing_packages_from_input(
+        self, packages, poetry_content, target_section
+    ):  # type: (List[str], Dict, str) -> List[str]
+        existing_packages = []
+
+        for name in packages:
+            for key in poetry_content[target_section]:
+                if key.lower() == name.lower():
+                    existing_packages.append(name)
+                    self.line(
+                        "{name} is already in pyproject.toml. Skipping.".format(
+                            name=name
+                        )
+                    )
+        packages = [name for name in packages if name not in existing_packages]
+
+        return packages
```

Comment:
Would be great if we can make this a filter to split out existing and new packages so that we can, at the end of the command execution (if not quiet) notify the use that these packages already exists and at version x, and hence were skipped.

The final message should also probably indicate that if they want to update they should use either update or `poetry add package@version syntax`.